api/openapi_server/__main__.py

```diff
@@ -1,105 +1,15 @@
-    parser.parse()
-
-    return parser.specification
-
-def get_version():
-    from importlib.metadata import version, PackageNotFoundError
-
-    try:
-        return version("homeuniteus-api")
-    except PackageNotFoundError:
-        # package is not installed
-        return "0.1.0.dev0"
-
-def create_app():
-    '''
-    Creates a configured application that is ready to be run.
-
-    The application is configured from external environment variables stored
-    in either a local `.env` in a development environment or from environment
-    variables that are configured prior to running the application.
-
-    This function is made available at the module level so that it can
-    be loaded by a WSGI server in a production-like environment with
-    `openapi_server.__main__:create_app()`.
-    '''
-
-    # `connexion.App` is aliased to `connexion.FlaskApp` (as of connexion v2.13.1)
-    # which is the connexion layer built on top of Flask. So we think of it as
-    # the connexion application.
-    connexion_app = connexion.App(__name__)
-    api_spec_path = connexion_app.get_root_path() / Path('./openapi/openapi.yaml')
-    parsed_specs = get_bundled_specs(api_spec_path)
-
-    parsed_specs['info']['version'] = get_version()
-
-    connexion_app.add_api(parsed_specs, pythonic_params=True)
-    connexion_app.add_error_handler(AuthError, handle_auth_error)
-
-    ENV_FILE = find_dotenv()
-    if ENV_FILE:
-        load_dotenv(ENV_FILE)
-    SECRET_KEY = env.get("SECRET_KEY")
-    env_config_profile = get_key(find_dotenv(), "CONFIG_PROFILE")
-
-    # The underlying instance of Flask is stored in `connexion_app.app`.
-    # This is an instance of `flask.Flask`.
-    flask_app = connexion_app.app
-    flask_app.json_encoder = encoder.JSONEncoder
-    flask_app.secret_key = SECRET_KEY
-
-    # Below, the Flask configuration handler is loaded with the
-    # application configuration settings
-    flask_app.config.from_object(compile_config(env_config_profile))
-
-    return connexion_app
-
+from openapi_server.app import create_app
 
 if __name__ == "__main__":
-    # This will/should be only run in a development environment
-
     connexion_app = create_app()
     flask_app = connexion_app.app
-
+    if flask_app.environment == "production":
+        raise EnvironmentError("The production configuration must be run on a "
+                               "production server. Connexion's app.run() method "
+                               "starts a development server that should be used "
+                               "for testing purposes only.")
     connexion_app.run(
-        host=flask_app.config[cp.HOST.name],
-        port=flask_app.config[cp.PORT.name],
-        debug=flask_app.config[cp.DEBUG.name],
-        use_reloader=flask_app.config[cp.USE_RELOADER.name],
+        host=flask_app.config["HOST"],
+        port=flask_app.config["PORT"],
+        load_dotenv=False
```

Comment:
I removed the `use_reloader` option since reloading is enabled when `FLASK_DEBUG` is true (link to [docs](https://flask.palletsprojects.com/en/2.3.x/config/#DEBUG)). I also did tried reloading code without issue on my machine.

We didn't have an option to control `use_debugger` before, and the development server has [a lot of other parameters](https://werkzeug.palletsprojects.com/en/2.3.x/serving/). I don't think it is necessary to add these options since they don't configure the application. They the configure the WSGI server.

If we find that we need to frequency modify the call to `.run` during development then it would be worth updating the development config.

Reply:
I remember Jules was having some kind of issue debugging when reloading was enabled and had to set `use_reloader` to `False` for his development. But if it works for you, then it should be ok to keep enabled.